Image upload

AladdinVR opened the pull request:
Getting policy from fastAPI server (see clearboard project [here](https://github.com/openfun/clearboard/pull/1) to see how it works ) and uploading image to backend server relatively to this policy

Commit:
Adding new package image-caputure its type for ts and declaring it

Commit:
new component that list all cameras of the device, it is called in JitsiComponent

Commit:
new component taking pictures from devices provided by CameraDetector

Commit:
getting roomName from JitsiComponent to CaptureImageComponent

Commit:
getting policy from fastapi server and sending image relativly to this policy

Commit:
getting policy from fastapi server and sending image relativly to this policy

sampaccoud commented:
@AladdinVR in your description you can use a link to the other project when you mention it

AladdinVR commented:
> @AladdinVR in your description you can use a link to the other project when you mention it

AladdinVR committed:
new package to send picture to backend server

AladdinVR committed:
new package to send picture to backend server

AladdinVR committed:
sending picture to backend server relatively to policy

AladdinVR committed:
sending picture to backend server relatively to policy

AladdinVR committed:
changing some parameters of the camera to maximise the quality of the picture

AladdinVR committed:
changing some parameters of the camera to maximise the quality of the picture

AladdinVR committed:
Capture image from webcam (#56)

* Adding new package image-caputure its type for ts and declaring it

* new component that list all cameras of the device, it is called in JitsiComponent

* new component taking pictures from devices provided by CameraDetector

Co-authored-by: aladdin <aladdin@MacBook-Pro-de-aladdin.local>

AladdinVR committed:
Revert "Capture image from webcam (#56)" (#60)

This reverts commit 81023e7a4572bceb6f1c6a95c0d9359eb84d2cff.

djexvr committed:
Refactor jitsi component (#62)

* 🏷️ (JitsiMeetAPI) Add options to config typing

Add config options from jitsi-meet documentation to type declaration
to allow the use of typing when using the api.

* ♻️ (JitsiMeetComponent) refactor jitsi component

Refactor component to expose the JitsiMeetAPI to be able to reuse
the functions in the different interfaces.

Remove the custom hangup button as it is possible to modify
the default button behavior through listener.

Add types for the api to make the component personalisation easier.

* ✏️ (types) fix typos in jitsi api config types

* 👽️ (JitsiComponent) replace deprecated config options

Some options from InterfaceConfig are deprecated and have been replaced
by their equivalents in config.

* ♻️  (Jitsi Frame) allow configuration of the jitsi frame

Rename the Jitsi Component to Jitsi Frame to be more explicit.

Allow the configuration of the Jitsi Frame to apply different options
and access Jitsi api through the configure prop.
This should simplify the creation of different interfaces.

Co-authored-by: Vianney Ruhlmann <vianney.ruhlmann@student-cs.fr>

djexvr commented on box-ui/src/components/ts/CameraDetectorComponent.tsx:
```diff
@@ -1,7 +1,7 @@
 import CaptureImage from './CaptureImageComponent';
 
-const CameraDetector = () => {
+const CameraDetector = (props: any) => {
```
You should add type for the props

AladdinVR committed:
Merge branch 'capture_image_from_webcam' of github.com:openfun/jitsi-box into capture_image_from_webcam

djexvr commented on box-ui/src/components/ts/CameraDetectorComponent.tsx:
```diff
@@ -1,7 +1,7 @@
+const CameraDetector = (props: any) => {
     const [cameraList, setCameraList] = useState([] as any[]);
     const selectCamera = () => {
         navigator.mediaDevices.enumerateDevices().then((devices: Array<any>) => {
```
You should use [existing types](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.mediadevices.html#enumeratedevices) for the devices

djexvr commented on box-ui/src/components/ts/CaptureImageComponent.tsx:
```diff
@@ -1,44 +1,59 @@
-    const start = (camera: string) => {
-        console.log(camera);
+    const start = (props: any) => {
+        console.log(props.camera['deviceId']);
```
You should remove console.log before merge

djexvr commented on box-ui/src/components/ts/CaptureImageComponent.tsx:
```diff
@@ -1,44 +1,59 @@
+                data.append('name', 'image');
+                data.append('file', blob, `${roomName}.png`);
+                // const address = `http://localhost:8070/policy?custom_address=${roomName}`; use this if you need to upload pictures on different servers (depending on the meeting for instance)
+                const address = 'http://localhost:8070/policy';
```
You should create a .env for this kind of url as it won't work on other devices

djexvr commented on box-ui/src/components/ts/CaptureImageComponent.tsx:
```diff
@@ -1,44 +1,59 @@
 import React, { useState, useEffect, useRef } from 'react';
 import { ImageCapture } from 'image-capture';
+import axios from 'axios';
+import FormData from 'form-data';
```
Why do you need a package for FormData since it is in the [JS api](https://developer.mozilla.org/fr/docs/Web/API/FormData) ?

AladdinVR committed:
rebase with origin pre_prod

AladdinVR committed:
adding typing

AladdinVR committed:
calling CameraDetector in MeetingComponent

AladdinVR committed:
missing new line at end of files

AladdinVR committed:
rebase

AladdinVR committed:
merge

AladdinVR committed:
typing

AladdinVR committed:
new type and update CaptureImageProps

AladdinVR committed:
rebase update

AladdinVR committed:
.env variables and readme

AladdinVR committed:
removed unused package
